### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by godrei**
This PR is the beginning of a fix for cache include and exclude path handling.

Since the current state of the code is not really unit testable, i will start by fixing this first, to be able to reproduce the issue with unit tests.

The current source uses the `CacheModel` struct's functions to solve the caching feature, every function modifies several fields of the `CacheModel`, which makes it really hard to follow what happens with this variables, what is modified where and so on...

This PR converts every struct function on `CacheModel` to regular functions, this way we can see which function receives and returns what.

In the next PR i will break these functions to better scoped smaller ones and try to reproduce the issue.

So in this PR you will see changes like 

`func (cacheModel *CacheModel) CreateTarArchive() error`

->

`func CreateTarArchive(compressArchive bool) (gzipWriter *gzip.Writer, tarWriter *tar.Writer, tarFile *os.File, err error)`

Check carefully if every function:

- accesses an appropriate variable instead of a CahceModel.Field
- returns a variable instead of modifying a CahceModel.Field

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -72,67 +72,8 @@ type CacheModel struct {
...
-		CompressArchive:     os.Getenv("compress_archive"),
-		FingerprintMethodID: os.Getenv("fingerprint_method"),
-		Paths:               os.Getenv("cache_paths") + "\n" + os.Getenv("bitrise_cache_include_paths"),
-		IgnoredPaths:        os.Getenv("ignore_check_on_paths") + "\n" + os.Getenv("bitrise_cache_exclude_paths"),
-	}
-}
-
 // NewCacheModel ...
-func NewCacheModel(configs *ConfigsModel) *CacheModel {
+func NewCacheModel(configs *Config) *CacheModel {
```
Comment (BirmacherAkos) [inline main.go]: Why do we use pointer here (*Config), instead of the default copy?
This method will not change anything in the configs.

File: main.go [hunk 2]
```diff
@@ -188,53 +127,38 @@ func (cacheModel *CacheModel) CloseTarArchive() error {
...
 		return err
 	}
 
-	if _, err := io.Copy(cacheModel.TarWriter, bytes.NewReader(filePathMapBytes)); err != nil && err != io.EOF {
+	if _, err := io.Copy(tarWriter, bytes.NewReader(filePathMapBytes)); err != nil && err != io.EOF {
 		return err
 	}
 
-	if err := cacheModel.TarWriter.Close(); err != nil {
+	if err := tarWriter.Close(); err != nil {
```
Comment (BirmacherAkos) [inline main.go]: I think a `defer func() { .close }` would be better for the Writer / File close.

**Comment by BirmacherAkos** [comment]
Please check / fix my notes in your next PR.

File: main.go (hunk 2 above)
Comment (trapacska) [reply main.go]: @BirmacherAkos There will be no any extra job for tarwriter, in case when you know that you can close a writer just close it straight away. However, if you are expecting something to happen which block you later on to close the writer then in that cases you need to use defer.

**Comment by godrei** [comment]
closing as i want to merge this PR into the development branch.
